source/extensions/filters/http/compressor/compressor_filter.cc

```diff
@@ -322,7 +323,7 @@ CompressorFilter::chooseEncoding(const Http::ResponseHeaderMap& headers) const {
   }
 
   // Find all compressors enabled for the filter chain.
-  std::map<std::string, uint32_t> allowed_compressors;
+  std::map<std::string, std::pair<uint32_t, bool>> allowed_compressors;
```

Comment:
nit: can you factor out the pair into a struct with named members? It's really hard to read the code and understand what is going on. I think it would be much more readable that way. Thank you.

Reply:
OK, done, thanks!